test/rss.cpp

```diff
@@ -154,6 +154,41 @@ TEST_CASE("W3CDTF parser behaves correctly") {
 		REQUIRE(rsspp::rss_parser::__w3cdtf_to_rfc822("2008-12-30T10:03:15-08:00") == "Tue, 30 Dec 2008 18:03:15 +0000");
 	}
 
+	// There has been a problem in the C date conversion functions when the TZ is
+	// set to different locations, and localtime is in daylight savings. One of
+	// these two next tests should be in active daylight savings
+	SECTION("W3CDTF with -08:00 timezone Pacific") {
+		setenv("TZ", "US/Pacific", 1);
+		tzset();
+		REQUIRE(rsspp::rss_parser::__w3cdtf_to_rfc822("2008-12-30T10:03:15-08:00") == "Tue, 30 Dec 2008 18:03:15 +0000");
+		unsetenv("TZ");
```

Comment:
I think tests should *preserve* the environment, not clear it, because clearing prevents us from passing `TZ` from the outside to the tests that happen to run after this one. So instead of unsetting `TZ`, please set it back to whatever value it previously had.

You can also DRY this block of tests by using so-called "sections" that our testing framework provides. Take a look at [a test](https://github.com/akrennmair/newsbeuter/blob/b6383a64cdfed333e2e31a33f908b33dd42ae3da/test/textformatter.cpp#L95-L119) that already uses them, and also [the framework's docs](https://github.com/philsquared/Catch/blob/master/docs/tutorial.md#test-cases-and-sections).